pymeasure/experiment/workers.py

```diff
@@ -115,10 +118,47 @@ def emit(self, topic, record):
         except (NameError, AttributeError):
             pass  # No dumps defined
         if topic == 'results':
-            self.recorder.handle(record)
+            self.handle_record(record)
+        elif topic == 'batch results':
+            self.handle_batch_record(record)
         elif topic == 'status' or topic == 'progress':
             self.monitor_queue.put((topic, record))
 
+    def handle_record(self, record: dict[str, Any]) -> None:
+        self.recorder.handle(record)
+
+    def handle_batch_record(self, record: Any) -> None:
```

Comment:
Currently, I think it is policy not to include type hints (also holds for _is_dictionary_of_sequences)

It also seems to be the cause of the failing CI (as `dict[...]`-style typing is not supported in python 3.8)

Reply:
Do you want me to remove (any) type hints? Personally I think it is always good to provide type hints as most editors can then already detect wrong usage of certain functions. It also makes it clearer to a user what they can expect from a function.

I believe I can fix the CI for py3.8 by using `Dict[str, Any]` instead of `dict`.

Reply:
Fixed the typing for python 3.8, but let me know if you want me to remove it.

Reply:
Actually, some more recent Tools do not like these old typings.
Better is to import annotations from __future__, afterwards typing works even for older Python.

Originally (way back), it was decided against type hints. Now, little speaks against it, except, that the code base does not use them yet.

Reply:
I always forget about the `__future__` imports, changed the typing to use e.g. `dict[str, Any]` instead of `Dict[str, Any]`. Also took the liberty to specify the types for the other functions in `workers.py` where that made as a step towards improving the code base.

Reply:
Sounds good to me; I'm generally in favour of type hints, so this sits fine with me.